### PR TITLE
dnsmasq: remove 400-NAK, tags, static dhcp, firewall rule, allow autoritative and manualy created interfaces !

### PR DESCRIPTION
**Pull request by sduponch**

### CODE REVIEW

File: dnsmasq/files/dnsmasq.init [hunk 1]
```diff
@@ -455,6 +460,8 @@ dhcp_add() {
 		DNS_SERVERS="$DNS_SERVERS $dnsserver"
 	}
 
+	dnsmasq_device_list="${dnsmasq_device_list} $ifname"
+	dnsmasq_interface_list="${dnsmasq_interface_list} $net"
```
Comment (gregdel) [inline dnsmasq/files/dnsmasq.init]: Where is it used ?

Comment (sduponch) [inline dnsmasq/files/dnsmasq.init]: - dnsmasq_device_list is used to list network devices used by dnsmasq to set dhcp-ignore on all other device like loopback, tun0, xtun0...
```
dhcp-host=a4:5e:60:c4:c8:89,set:ovh1,192.168.0.25,astreinte

dhcp-range=set:lan,192.168.0.100,192.168.0.249,255.255.255.0,12h
dhcp-option=lan,43,01:04:A7:72:F2:2A
dhcp-option=lan,option:router,192.168.0.20
dhcp-option=lan,option:dns-server,192.168.0.20

dhcp-range=ovh1,192.168.0.1,static,255.255.255.0,12h
dhcp-option=ovh1,option:router,192.168.0.251
dhcp-option=ovh1,option:dns-server,8.8.8.8

no-dhcp-interface=eth0
no-dhcp-interface=if0
no-dhcp-interface=if1
no-dhcp-interface=lo
no-dhcp-interface=ovh2
no-dhcp-interface=tun0
```

- dnsmasq_interface_list is used to list interfaces where a dhcp is configured, that allow to setup a static dhcp when a host is tagged with an interface that is not configured.

```
config dhcp 'lan'
	option interface 'lan'
	option start '100'
	option limit '150'
	option leasetime '12h'
	list dhcp_option '43,01:04:A7:72:F2:2A'

config dhcp 'wan'
	option interface 'wan'
	option ignore '1'

config host
	option name 'astreinte'
	option mac 'a4:5e:60:c4:c8:89'
	option ip '192.168.0.25'
	option tag 'ovh1'
```

Comment (gregdel) [inline dnsmasq/files/dnsmasq.init]: You don't get my question, where is is variable used in the dhcp_add function

Comment (sduponch) [inline dnsmasq/files/dnsmasq.init]: This used after the dhcp_add, Line 1115 and 1124

Comment (sduponch) [inline dnsmasq/files/dnsmasq.init]: This is a list variable that call itself and this is used later at line 1115 and 1124

Comment (gregdel) [inline dnsmasq/files/dnsmasq.init]: > This used after the dhcp_add, Line 1115 and 1124
You mean line 463 / 464 right ? This is where is put my comment, so I can see these lines, but I don't see this variable used in the dhcp_add function

**Comment by sduponch** [comment]
Changes in y last amend is to append a RETURN rules in last dnsmasq iptables to allow unknown user to reach the DHCP.

**Comment by sduponch** [comment]
My PR need a little work on luci-mon-adminnetwork to display correct gateway list (as i used dhcp tags list that is no longer necessary), but everyting works as expected on V2A.

File: dnsmasq/files/dnsmasq.init (hunk 1 above)
Comment (sduponch) [inline dnsmasq/files/dnsmasq.init]: Initialized at 1071, 1072
Populated in dhcp_add at 463, 464       
used at 1115, 1125

**Pull request closed without merging**